### ISSUE TITLE
Proposal: use Restangular instead of $http in services

### ISSUE THREAD

**Issue by tiltec** [yunity/foodsaving-frontend#81]
[Restangular](https://github.com/mgonto/restangular) seems to be quite popular (over 7000 stars on GitHub) and promises to remove a lot of redundancy from the service definitions.
Also, it seems to be easy to enable site-wide caching.

I'm collecting more upsides/downsides. Obviously, the http part seems to be the easy part of writing a webapp, and I think I could simplify the code even more.

**Comment by Sneezoo** [comment]
It looks like there is no active development for it anymore... And I think it's not valueable changing all our services for this. The outcome seems to be really small...

**Comment by D0nPiano** [comment]
I think it could be really nice to have a caching option though (of course it doesn't need to be with this library), for things like:

>  I saw that opening http://localhost:3000/#!/group/1 fires 5 requests to /api/stores/2/ directly after each other.. I'll try to figure out why that happens and find a solution.

(Every pickuplist item needs to know the store name and fires another request to the backend)

**Comment by Sneezoo** [comment]
Rather fix the backend than firing tons of requests... That's supposed to be an issue in https://github.com/yunity/foodsaving-backend.. and it already has been one.
https://github.com/yunity/foodsaving-backend/issues/194

>I'm strongly in favour of a shallow representation to keep the backend simple. Since we are using >HTTP2, additional requests should be pretty cheap and can even be run in parallel.
>
>So I would recommend to GET all member details separately. Or get all user details in one go.

I think it's weird behaviour of a single page application to make requests for each one *name* string.

About caching:
a) browsers already use HTTP caching for Ajax requests
b) caching **may** result to unexpected behaviour for example when trying to refresh some data (=> no refresh as data is loaded from cache)
c) for now we don't have any performance issues. And before using special techniques or whatever we should try to reduce the amount of requests and fix any performance issues from architecture, code, ...

**Comment by nicksellen** [comment]
I would oppose adding Restangular, it adds complexity, and another library to understand. There are really not many endpoints yet...

I think it is too early to worry much about optimising requests and it won't be difficult to do once needed. Much more tricky and important right now is actually getting a usable interface working.

**Comment by tiltec** [comment]
You all have good points, but I still think that we would be better off in the long run if we would abstract away some of the http stuff.

**Comment by nicksellen** [comment]
In the long run anything could happen :) Libraries come and go. Close the issue then? Unless any short term change is suggested?

**Comment by tiltec** [comment]
Yes, regardless of reasons - if there is strong resistance, I'm willing to propose this later again (will do a list of those things ^^)
